forge-gui-desktop/src/main/java/forge/view/arcane/CardPanel.java

```diff
@@ -299,6 +301,13 @@ protected final void paintComponent(final Graphics g) {
             g2d.fillRoundRect(cardXOffset - n2, (cardYOffset - n2) + offset, cardWidth + (n2 * 2), cardHeight + (n2 * 2), cornerSize + n2, cornerSize + n2);
         }
 
+        // Yellow border for cards in hand that can be cast with available mana
+        if (canCastCard() && isPreferenceEnabled(FPref.UI_SHOW_CASTABLE_BORDER)) {
```

Comment:
you should change the order so the expensive check isn't run when the preference is off